src/languages/gherkin.js

```diff
@@ -5,66 +5,146 @@
  Website: https://cucumber.io/docs/gherkin/
  */
 
-export default function(hljs) {
+export default function() {
   return {
     name: 'Gherkin',
     aliases: [ 'feature' ],
-    keywords: {
-      // Custom pattern to support keywords with spaces and those ending with a colon
-      $pattern: /[A-Z][a-z]+(?: [A-Z][a-z]+)?:?/,
-      // Add positive lookbehind to ensure keywords are at the beginning of a line
-      // $pattern: /(?<=^[ \t]*)[A-Z][a-z]+(?: [A-Z][a-z]+)?:?/,
-      keyword: [
-        'Feature:',
-        'Rule:',
-        'Example:', 'Scenario:',
-        'Given', 'When', 'Then', 'And', 'But',
-        'Background:',
-        'Scenario Outline:', 'Scenario Template:',
-        'Examples:', 'Scenarios:'
-      ]
-    },
     contains: [
       {
-        className: 'keyword',
-        begin: '\\*',
-        relevance: 0
+        begin: [
+          /^(Feature|Rule|Examples?|Scenario(?:s| Outline| Template)?|Background)/,
+          /:/
+        ],
+        beginScope: {
+          1: 'keyword',
+          2: 'punctuation',
+        },
+        end: /$/,
+        contains: [
+          {
+            scope: 'variable',
+            begin: /<[^>\s]+>/
+          },
+        ]
       },
       {
-        className: 'meta',
-        begin: '@[^@\\s]+'
+        begin: /^(?:Given|When|Then|And|But)\b/,
+        beginScope: 'keyword',
+        end: /$/,
+        contains: [
+          {
+            scope: 'variable',
+            begin: /<[^>\s]+>/
+          },
+        ]
       },
       {
-        begin: '\\|',
-        end: '\\|\\w*$',
+        begin: /^\*(?=[ \t])/,
+        relevance: 0,
+        beginScope: 'keyword',
+        end: /$/,
         contains: [
           {
-            className: 'string',
-            begin: '[^|]+'
-          }
+            scope: 'variable',
+            begin: /<[^>\s]+>/
+          },
         ]
       },
       {
-        className: 'variable',
-        begin: /<[^>\s]+>/
+        scope: 'meta',
+        begin: '@[^@\\s]+'
+      },
+      {
+        scope: 'comment',
+        begin: /^#/,
+        end: /$/
       },
-      // Comments can only start at the beginning of a line
-      hljs.COMMENT(/^[ \t]*#/, /$/),
-      // Use positive lookbehind (once available) to exclude leading spaces from comment
-      // hljs.COMMENT(/(?<=^[ \t]*)#/, /$/),
       {
-        className: 'string',
+        scope: 'string',
         variants: [
           {
-            begin: /"""/,
+            begin: /^"""/,
             end: /"""/
           },
           {
-            begin: /```/,
+            begin: /^```/,
             end: /```/
           }
+        ],
+      },
+      {
+        begin: /^\|.*\|$/,
+        scope: 'string'
+      },
+      {
+        begin: /^[ \t]+/,
```

Comment:
This repetition is frustrating.  Could a single top-level rule to eat tabs at the start of lines accomplish this same thing?  If not why not add it to each multi-match rule as an optional first component:

```
        begin: [
          LINE_BEGINS_WS,
          /(Feature|Rule|Examples?|Scenario(?:s| Outline| Template)?|Background)/,
          /:/
        ],
        beginScope: {
          2: 'keyword',
          3: 'punctuation',
        },
```

Reply:
Yes, eating whitespace of the beginning of the line with an empty mode seems to work. Is that what you had in mind? 😄

Reply:
The multi-match method I show above is what I had in mind, yes.  That's the best way I know to handle this kind of thing (without putting the whitespace inside the highlight zone).